### DISCUSSION TITLE
Link to alternative project for Tuya

### DISCUSSION THREAD

**Discussion by matejsp** [arendst/Tasmota#17444] [Show and tell]
Since Tasmota will never support RTL8710BN and other Tuya modules, according to documentation it would be nice to link to alternative project that are focused on this. https://tasmota.github.io/docs/Tuya-Convert/

https://github.com/kuba2k2/libretuya
https://github.com/openshwprojects/OpenBK7231T_App

**Comment by sfromis** [comment]
Docs pages have edit links for people to proprose changes, Wiki-style, nothing to open an "issue" (which is for bugs) about :smile:

**Reply by zombieregime** [reply]
Yeah, its kind of silly all these people who cant wrap their head around the concept that Tasmota was intended to work on ESP based devices, written by someone who (probably) has the most meaningful experience with ESP MCUs, at a time when Tuya was only using ESP chips, and here is the best part, FOR FREE!!!! But because Tuya up and sources a different MCU for their products that means the author and devs of Tasmota deserve to be treated like they are being unreasonable, when the people complaining havent given the slightest consideration towards the difference in architecture, communication buses, or (freaking gasp) learning to code for the other architecture and contributing to the community of jailbreakers they are relying on for the functionality they expect. "Learn something?! I just want this whatsit from that multinational multimillion dollar company that has a vested interest in protecting their business model to be easily reprogrammed and 'just work' from my phone for free in 3 taps or less! AND IF I CANT DO THAT SOMEONE ELSE IS BEING LAZY!!!" Those types of people, most likely (at least thats how they appear to everyone else who bothered to learn something about electronics instead of copypastaing random code on the internet, then demand someone fix it for them since after all they did hold a gun to that persons head and force them to run homebrew code to overwrite the firmware on something that in no way matched the pictures....so its the authors fault.....). 

The maker movement has been both blessing and curse. The ridiculously low cost and readily available components these days still makes my head spin. Motorized linear potentiometers were unheard of if you didnt build high end studio mixer boards, now they're 5 for $20 on amazon! Absolutely unbelievable.... And dont get me started on steppers and servos. However the other side of that is now there are tons of people who think they're hot s#!t in a champagne glass because they can plug a few wires into holes, hit ctrl+c, then ctrl+v, and make something move. But since no one has posted a scratch file on git for controlling the thing they just make a insta post with some stupidly inaccurate title and soak in the sycophantic praise for doing a thing while they move on to copying the next 'top 5 (same old garbage) things to do with a raspberry pi', as cold diarrhea in a dixie cup is want to do... I swear every time I see someone hooking up an i2c device up to an arduino which is then hooked up to a serial to USB cable to a RPi which is then accessed over the command line or VNC I wand to freaking scream and kick a kitten. They know so little of what they are doing they cant see how convoluted it all is and are in no way inspired to (at least) translate the arduino C code into python on the rpi because its not in the tutorial. Or freaking plug the serial USB cable into the desktop! (Picard facepalm and grumbles) ... OK, I should stop, but I could circle this point all night......

Yes, I feel strongly about this subject. No, my mind cannot be changed. Also no, I am not fun at parties (anymore).

PS - For the record, I am thankful for the work that has been put into Tasmota. Thumbs up to the original author and all the devs who helped make it better over the years!

PSS - (pre/post-face: all the "you"s in this post have been generalized proverbial colloquial "you"s, not pointed direct "you"s to the thread Im replying under) If you're new and at the point where you're doing convoluted stuff like that last bit, hey its cool. We have all been there. Dont worry about it. But for the love of pasta, experiment, explore, use what you have learned and do something not in a tutorial. And dont be afraid to google command syntax, even seasoned pros still do it to make sure they have the code grammar correct. Just dont stagnate! You can do it, I believe in you!

**Comment by hemangjoshi37a** [comment]
Hey brother. Can you please checkout my discussion #17815 please.

**Reply by matejsp** [reply]
Sry, but I don't understand how is this related to this topic? Do you support new Tuya modules?

**Reply by sfromis** [reply]
It is just spam. Maybe not commercial, but still spam.

**Reply by hemangjoshi37a** [reply]
@matejsp we are going to support tuya in next month. 

@sfromis definitely not spam, but very useful app to everyone having tasmota project.

**Reply by matejsp** [reply]
How will you support latest tuya if tasmota is not supporting latest tuya?
Do you have any github code that we can check? I really don't want to install some random third party app.

It seems that you are selling this solution or at least development part.
https://hjlabs.in/product/android-ios-web-linux-osx-windows-app-development-with-flutter-firebase-stack/

**Reply by hemangjoshi37a** [reply]
Who said I am working for free? At the end of the day I need to purchase the dinner to eat and for that I need to earn something obviously also you have to do so. Also my tasmota app is totally free and has nothing to do with my paid app development services.

Also we are working on some tuya project that might be merged in the future. 

I don't know why are you so defensive brother.

**Reply by sfromis** [reply]
Your desire to earn money too means that you barging in to promote your product is commercial spam, not just topic spam. That's the way to make users **not** want to use your product.

**Reply by hemangjoshi37a** [reply]
Are you out of your mind? I am saying that I am offering free app. Does that seem any commercial to you? I don't know man.

And I am also telling that my other services has nothing to do with this FREE app. If you can NOT understand that this is any commercial spam than I don't know brother you need some help.

**Reply by matejsp** [reply]
I understand your urge to promote you free, useful to some, closed source app (I use free and open source HA app with Tasmota). However my discussion is not the place since your solution adds zero value to the discussion. Please stop!

Now if you have anything to share regarding your Tuya project with support for RTL8710BN, I would be more than happy to have a conversation about it.
